### ISSUE TITLE
False positive for `Layout/HeredocIndentation` with `parser_prism`

### ISSUE THREAD

**Issue by andyw8** [rubocop/rubocop#12869]
Similar to https://github.com/rubocop/rubocop/issues/12787:

```
<<~TXT.squish
  a
  b
TXT
```

Prism v0.27.0, RuboCop v1.63.3

cc @kddnewton

**Comment by andyw8** [comment]
Another variant:

```
<<~RUBY, "y")
  a
  b
RUBY
```

**Comment by emptyflask** [comment]
With blocks as well:
```ruby
Rails.logger.debug { <<~LOG }
  a
  b
LOG
```

**Comment by frederikspang** [comment]
Also reproduces on prism 0.29.0.
